parm/snow/obs/config/sfcsno_snow.yaml.j2

```diff
@@ -5,9 +5,11 @@
       halo size: 250e3
     obsdatain:
       engine:
-        type: bufr
+        type: script
+        script file: "{{ PARMgfs }}/gdas/snow/ioda_bufr_python_encoder.py"
+        args:
+          input_path: '{{ DATA }}/obs/{{ OPREFIX }}sfcsno.tm00.bufr_d'
         obsfile: '{{ DATA }}/obs/{{ OPREFIX }}sfcsno.tm00.bufr_d'
```

Comment:
Don't forget to add an issue so they do not assume the formatting of the YAML file (obsfile variable shoould not be necessary). Reading the backend engines YAML breaks encapsulation (external modules should not be dependent on the private configuration for a backend type. Some backends may not have any kind of file associated with them (like a db...).

Reply:
yes thanks @rmclaren I'm aware of that issue, I/we/others need to figure out how best to handle it on the workflow side.